nixos/modules/services/video/broadcast-box.nix

```diff
@@ -0,0 +1,189 @@
+        RestartSec = "10s";
+      };
+    };
+
+    networking.firewall = mkIf cfg.openFirewall {
+      allowedTCPPorts = [ cfg.tcp.port ] ++ optional cfg.ssl.redirect 80;
+      allowedUDPPorts = optional (cfg.udp.port != null) cfg.udp.port;
+    };
+  };
+
+  options.services.broadcast-box = {
+    enable = mkEnableOption "broadcast box";
+    package = mkPackageOption pkgs "broadcast-box" { };
+
+    listenAddress = mkOption {
+      type = types.str;
+      default = "";
+      example = "127.0.0.1";
+      description = mdDoc ''
+        Address the HTTP server will listen on.
+      '';
+    };
+
+    tcp.port = mkOption {
+      type = types.port;
+      default = 8080;
+      description = mdDoc ''
+        TCP Port the HTTP server will listen on.
+      '';
+    };
+
+    udp.port = mkOption {
+      type = types.nullOr types.port;
+      default = null;
+      example = 3000;
+      description = mdDoc ''
+        UDP port to serve all WebRTC traffic over. By default, a random UDP
+        port is selected which `openFirewall` will **not** open.
+      '';
+    };
+
+    udp.interfaceFilter = mkOption {
+      type = types.str;
+      default = "";
+      example = "lo";
+      description = mdDoc ''
+        Only use this interface for UDP traffic.
+      '';
+    };
+
+    openFirewall = mkOption {
+      type = types.bool;
+      default = false;
+      description = mdDoc ''
+        Whether to open the specified ports in the firewall. Also opens port 80
+        if `ssl.redirect` is enabled.
+
+        If `udp.port` is `null`, it is assigned a random port, so this option
+        will **not** open it in the firewall.
+      '';
+    };
+
+    autoStart = mkOption {
+      type = types.bool;
+      default = true;
+      description = mdDoc ''
+        Whether Broadcast Box should be started automatically.
+      '';
+    };
+
+    networkTestOnStart = mkOption {
+      type = types.bool;
+      default = false;
+      description = mdDoc ''
+        Whether to run a network test on start. Broadcast Box will exit if the
+        test fails.
+      '';
+    };
+
+    stunServers = mkOption {
+      type = types.listOf types.str;
+      default = [ ];
+      description = mdDoc ''
+        List of STUN servers. Useful if Broadcast Box is running behind a NAT.
+      '';
+    };
+
+    ssl.redirect = mkOption {
+      type = types.bool;
+      default = false;
+      description = mdDoc ''
+        Whether to run an HTTP server on port 80 that redirects HTTP traffic to
+        HTTPS.
+      '';
+    };
+
+    ssl.cert = mkOption {
+      type = types.nullOr types.path;
+      default = null;
+      description = mdDoc ''
+        Path to the SSL certification for Broadcast Box's HTTP server.
+      '';
+    };
+
+    ssl.key = mkOption {
+      type = types.nullOr types.path;
+      default = null;
+      description = mdDoc ''
+        Path to the SSL key for Broadcast Box's HTTP server.
+      '';
+    };
```

Comment:
I know that you're simply copying the name of the environment variables here but I feel like putting the SSL options in the namespace `https.` would make this a bit easier to understand. No requirement, though, just a thought since the naming comes from upstream.

Reply:
That's a good idea, I'll group them under `https` along with the `redirect` option.